packages/neon/neon/lib/src/pages/login_check_account.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+import 'package:neon/l10n/localizations.dart';
+import 'package:neon/src/bloc/result.dart';
+import 'package:neon/src/bloc/result_builder.dart';
+import 'package:neon/src/blocs/accounts.dart';
+import 'package:neon/src/blocs/login_check_account.dart';
+import 'package:neon/src/models/account.dart';
+import 'package:neon/src/router.dart';
+import 'package:neon/src/widgets/account_tile.dart';
+import 'package:neon/src/widgets/exception.dart';
+import 'package:neon/src/widgets/linear_progress_indicator.dart';
+import 'package:neon/src/widgets/validation_tile.dart';
+import 'package:provider/provider.dart';
+
+class LoginCheckAccountPage extends StatefulWidget {
+  const LoginCheckAccountPage({
+    required this.serverURL,
+    required this.loginName,
+    required this.password,
+    super.key,
+  });
+
+  final String serverURL;
+  final String loginName;
+  final String password;
+
+  @override
+  State<LoginCheckAccountPage> createState() => _LoginCheckAccountPageState();
+}
+
+class _LoginCheckAccountPageState extends State<LoginCheckAccountPage> {
+  late final LoginCheckAccountBloc bloc;
+
+  @override
+  void initState() {
+    super.initState();
+
+    bloc = LoginCheckAccountBloc(
+      widget.serverURL,
+      widget.loginName,
+      widget.password,
+    );
+  }
+
+  @override
+  void dispose() {
+    bloc.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(final BuildContext context) => Scaffold(
+        appBar: AppBar(
+          leading: const CloseButton(),
+        ),
+        body: Center(
+          child: Padding(
+            padding: const EdgeInsets.all(10),
+            child: ConstrainedBox(
+              constraints: const BoxConstraints(
+                maxWidth: 640,
+              ),
```

Comment:
Something for the new NeonTheme?

Reply:
What's the best way to add it there?

Reply:
I haven't seen full constraints in ThemeData yet and it looks like flutter would rather just store the value (`double 640`) but imo storing the entire constraint would also be ok.

---

For now it might be ok just to add it as a field in the NeonTheme.
You'll need to both add it to the copyWith  and lerp function so flutter will automatically animate transitions.
You can read more at: https://api.flutter.dev/flutter/material/ThemeExtension-class.html
It could look like:
[patch.txt](https://github.com/provokateurin/nextcloud-neon/files/11995166/patch.txt)

---

I think eventually we should organize our fields in the NeonTheme in sub classes (similar to what the flutter ThemeData does with TextTheme, ButtonTheme ...) but we can also change this once we start adding more fields to it.

Reply:
Ah and you can access it through `Theme.of(context).extension<NeonTheme>()!` everywhere where we have the context.

Reply:
Thanks a lot for your explanation! I already read a bit into ThemeExtensions, but I just wanted to make sure that I do it correctly and don't waist time implementing it the wrong way

Reply:
And please add doc comments to the property in the NeonTheme.

Ideally the NeonTheme should be exported eventually so other apps can make use of it.